Tasks/AzurePowerShellV4/TryMakingModuleAvailable.ps1

```diff
@@ -0,0 +1,78 @@
+[CmdletBinding()]
+param (
+    [string]
+    $targetVersion,
+
+    [string] [Parameter(Mandatory = $true)] [ValidateSet("Windows", "Linux", "Mac")]
+    $platform
+)
+
+try {
+    $isWin = $platform -eq "Windows"
```

Comment:
do we really need to pass in a platform argument ? I believe the agent already provides you with a variable that tells you platform it is running on. the variable is called 'Agent.os' which is available as an environment variable like so: $env:AGENT_OS

Reply:
I don't see any harm in the way it is currently as we are doing similar thing already like having a separate file InitializeAz.ps1 in case of linux and CoreAz.ps1 in case of windows.